### PR TITLE
Interactions: Add clear-completed-tasks button, update local storage completed boolean value.

### PR DESCRIPTION
**Pull request by Ellon-M**
## Changes in this PR

-  Added a `completed()` method in the `ManageList` class to check if the to-do tasks are completed, and update them accordingly in the browser's local storage.
- Added a button to clear all the completed tasks from the list.

### CODE REVIEW

File: src/js/manageList.js [hunk 1]
```diff
@@ -85,29 +93,43 @@ export default class ManageList {
...
+      }
+    })
+  }
+
+  clearCompleted() {
+    let retreived = JSON.parse(localStorage.getItem('tasks'));
+
+    const filtered = retreived.filter(task => task.completed !== true);
+    localStorage.setItem('tasks', JSON.stringify(filtered));
   }
```
Comment (aimalamiri) [inline src/js/manageList.js]: Good job! but according to project requirements, we expect you to put this method in a new file, and import it as a module for better code isolation. Kindly make sure to export this to a new file.

image

**Review by aimalamiri — CHANGES_REQUESTED** [review]
Hi @Ellon-M, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlmostThereReachGIF

## Highlights

- The app is designed very well and look fantastic ✔️ ⭐
- Good job for using the right GitHub Flow 💯
- Awesome file organization 🗃️
- Commits are descriptive and meaningful 👍
- Linters are properly set up and passing 🟢 👍

## Required Changes ♻️

**When the user clicks on the checkbox it completes the task, but after refreshing the page the user can't see which tasks are completed, although it's already updated in the local storage, please make sure to fix this bug.**

image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/manageList.js (hunk 1 above)
Comment (Ellon-M) [reply src/js/manageList.js]: Well noted. All the changes have been made.

File: src/js/manageList.js [hunk 2]
```diff
@@ -85,29 +93,43 @@ export default class ManageList {
...
+        desc.classList.remove('striked');
+      }
+    })
+  }
+
+  clearCompleted() {
+    let retreived = JSON.parse(localStorage.getItem('tasks'));
+
+    const filtered = retreived.filter(task => task.completed !== true);
+    localStorage.setItem('tasks', JSON.stringify(filtered));
```
Comment (omarsalem7) [inline src/js/manageList.js]: - [ ] As the previous reviewer stated, kindly put this method in a new file, and import it as a module for better code isolation

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @Ellon-M , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

Almost near

 ## Highlights
✔️ All linter checks passed
✔️ Descriptive pull request

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] [OPTIONAL] It will be better if you add a future features section in your readme to let others know what feature will be working on it in the future.  

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/manageList.js (hunk 2 above)
Comment (Ellon-M) [reply src/js/manageList.js]: Done. Please let me know if there is anything else left out.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🍾🎉**
Hello @Ellon-M 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**